### ISSUE TITLE
Create a CHS-jsPsych video-consent plugin

### ISSUE THREAD

**Issue by becky-gilbert** [lookit/lookit-api#1295]
# TL;DR

We need to create a custom CHS-jsPsych plugin for recording consent videos.

# Narrative

As a researcher using CHS, I need my participants to record a consent video. This requires displaying the consent statement and other information on the page, along with the participant's webcam feed. I also need this page to include the ability for participants to start recording, stop recording, and play back the video. Ideally I should not have to do much programming to get this to work, or even have to copy/paste large chunks of code.

As a CHS family, I would like the consent recording process to be clear and easy. I would like to be able to watch my recording and re-record it if necessary. Ideally it would the consent recording process would look and work similarly across experiments.

# Acceptance Criteria

- [x] Tests
- [x] Scoping: decide exactly how similar it needs to be to the [EFP video-consent frame](https://lookit.readthedocs.io/projects/frameplayer/en/latest/components/exp-lookit-video-consent/doc.html) in terms of content/layout and parameters
- [x] Must include the ability to start/stop recording, re-record, and playback last recording
- [x] Must include a section for the to-be-read consent statement

# Implementation Notes

jsPsych has a [`mirror-camera` plugin](https://www.jspsych.org/7.3/plugins/mirror-camera/) for displaying the webcam feed back to the participant, as well as a [`record-video` extension](https://www.jspsych.org/7.3/extensions/record-video/) for creating video recordings. We should be able to make use of these existing features in our custom consent recording plugin in order to avoid re-inventing the wheel, and to take advantage of jsPsych's internal maintenance and updates. 

The video recording extension only allows the recording of a whole trial. We could consider collaborating with jsPsych to add parameters that allow the ability to link the recording start/stop/playback to page elements. Another option is to implement the recording entirely within the video-consent plugin, without using the jsPsych extension. With this approach, we might also find that it's possible to factor out some of the recording code so that it can be used as part of the [jsPsych plugin API](https://www.jspsych.org/7.3/reference/jspsych-pluginAPI/) as well as from within the extension.

**Comment by becky-gilbert** [comment]
Closed via https://github.com/lookit/lookit-jspsych/pull/52
